test/validation/api/time/time.c

```diff
@@ -276,12 +278,13 @@ static void time_test_monotony(void)
 	s_ns2 = odp_time_to_ns(ls_t2);
 	s_ns3 = odp_time_to_ns(ls_t3);
 
-	/* Time counting starts from zero. Assuming that the ODP instance has run
+	/* Time should not wrap in 10 years from ODP start. Assuming that the ODP instance has run
 	 * less than 10 minutes before running this test case. */
-	CU_ASSERT(ns1    < 10 * ODP_TIME_MIN_IN_NS);
-	CU_ASSERT(s_ns1  < 10 * ODP_TIME_MIN_IN_NS);
-	CU_ASSERT(l_ns1  < 10 * ODP_TIME_MIN_IN_NS);
-	CU_ASSERT(ls_ns1 < 10 * ODP_TIME_MIN_IN_NS);
+	limit = 10 * YEAR_IN_NS - 10 * ODP_TIME_MIN_IN_NS;
```

Comment:
I would drop the 10 min leeway. It is highly unlikely that the 10 min adjustment makes a difference. If it does makes a difference, maybe we would rather have the test fail when it should pass instead of pass when it should fail.

Actually, it is not entirely clear if the 10 year time period mentioned in the API means exactly 10 * 365 * 24 * 3600 seconds or 10 calendar years which can include 2 leap days and some leap seconds.

Reply:
Removed 10min margin in v2